fvm/evm/handler/blockstore.go

```diff
@@ -0,0 +1,59 @@
+package handler
+
+import (
+	gehtCommon "github.com/ethereum/go-ethereum/common"
+	"github.com/onflow/atree"
+
+	"github.com/onflow/flow-go/fvm/evm/types"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+var FlexLatestBlockKey = "LatestBlock"
+
+type BlockStore struct {
+	led         atree.Ledger
+	flexAddress flow.Address
+}
+
+var _ types.BlockChain = &BlockStore{}
+
+// NewBlockStore constructs a new block store
+func NewBlockStore(led atree.Ledger, flexAddress flow.Address) (*BlockStore, error) {
+	return &BlockStore{
+		led:         led,
+		flexAddress: flexAddress,
+	}, nil
+}
+
+// AppendBlock appends a block to the chain
+func (bs *BlockStore) AppendBlock(block *types.Block) error {
+	blockBytes, err := block.ToBytes()
+	if err != nil {
+		return types.NewFatalError(err)
+	}
+	err = bs.led.SetValue(bs.flexAddress[:], []byte(FlexLatestBlockKey), blockBytes)
```

Comment:
are we planning to keep older blocks? I guess here we just overwrite them

Reply:
not on chain.